enterprise/server/usage_service/usage_service.go

```diff
@@ -27,6 +31,9 @@ type usageService struct {
 }
 
 func New(env environment.Env) *usageService {
+	if !usage_config.UsageTrackingEnabled() {
```

Comment:
nit: same issue here re. `New*` returning nil -- would be slightly less surprising to do this check at the call site IMO

Reply:
The way I've seen this handled before is to instead call this method something like:

```
func Register(env environment.Env) {
  if enabled {
    env.SetUsageTracker(New(blah))
  }
}
```

This mostly works well -- the cases where it starts to break down are when you have many order-dependent backends and you need some set before or with others, then callers need to be aware of what's registered and not. That said, it forces you to break many of these dependencies and clarify things, which is kind of nice.